lib/tasks/change_allow_web_publish_to_false.rake

```diff
@@ -1,11 +1,38 @@
 #rake goodcity:change_allow_web_publish_to_false
-
 namespace :goodcity do
   desc 'change allow_web_publish to false if package.quantity = 0'
   task change_allow_web_publish_to_false: :environment do
     packages = Package.where("quantity = ? and allow_web_publish = ?", 0, true)
+    # code to create log for the rake
+    start_time = Time.now
+    rake_logger = Logger.new("#{Rails.root}/log/rake_log.log")
+    log = ("\n#{'-'*75}")
+    rake_logger.info(log)
+    log += ("\nRunning rake task 'change_allow_web_publish_to_false'....")
```

Comment:
Thanks Rajan, this is good and you've integrated the points I mentioned before. One thing that I've realised is that aside from reading logs on the console, we are using a machine log reader that looks at these logs. We've configured our logs to be machine readable using json key=value pairs and as such it would be better to have something more along those lines such as the following...

    # Logfile created on 2017-04-25 14:28:15 +0800 by logger.rb/47272
    I, [2017-04-25T14:28:22.282319 #4637]  INFO -- : Test

so something like

    rake_logger.info("task=change_allow_web_publish_to_false started")
    rake_logger.info("task=change_allow_web_publish_to_false Number of Packages to be affected =#{packages.count}"

etc etc

Chat to me online for more details.